Common/Extensions/UIViewController+Additions.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  ConfigureViewController.swift
+//  trello-app1
+//
+//  Created by Luis Felipe on 08/02/23.
+//
+
+import UIKit
+
+extension UIViewController {
+    
+    func configureViews(color: UIColor? = nil,
+                        collection collectionView: UICollectionView? = nil,
+                        scroll scrollView: UIScrollView? = nil) {
+        
+        view.backgroundColor = color
+        
+        guard let collectionView = collectionView else { return }
+        view.addSubview(collectionView)
+        NSLayoutConstraint.activate([collectionView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+                                     collectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+                                     collectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+                                     collectionView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)])
+        
+        guard let scrollView = scrollView else { return }
+        view.addSubview(scrollView)
+        NSLayoutConstraint.activate([scrollView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+                                     scrollView.widthAnchor.constraint(equalToConstant: 1000),
+                                     scrollView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+                                     scrollView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)])
+    }
```

Comment:
Aqui já da pra ter uma ideia do que foi comentado acima. Essa função adiciona para todos os view controllers que existem (ou que vierem a existir) a possibilidade de adicionar uma collection view e/ou um scroll view já definindo ali alguma ideia de restrições. Na minha visão isso se torna um uso bastante específico sendo liberada para todo mundo (potencial de confusão no futuro). Eu tiraria esse código daqui e deixaria ele por responsabilidade dos próprios view controllers mesmo. Analisaria o problema que originou a necessidade de colocar esse código a ser compartilhado aqui e veria se não tem uma outra forma de resolver. Senão imagina, em um cenário futuro, você ou alguém da equipe ta passando por um problema com a hierarquia de views em um view controller X e ta gastando tempo tentando entender a razão, aí pode ser difícil achar o ponto do código (se for em alguma dessas linhas) porque elas estão vazando do código que define o controlador que esta precisando do fix, quando acha o código nessa extensions pode ser tentado a resolver aqui, mas alterar o código aqui pode alterar o comportamento de outros n controllers na aplicação que escolherem usar essa implementação compartilhada. Tem que ter um cuidado aqui, pra no fim não cair numa armadilha. Nesse caso específico a gente pode sofrer igualzinho com o acoplamento imposto por herança e isso é porque o vilão na verdade não é a herança, o `XPTO: SuperClass`, é o compartilhamento implícito de implementação por meio do código.

Duas premissas que também podemos usar ao escrever e revisar código são: (i) não prover implementação compartilhada se ela não puder ser aplicada por todas as instâncias/objetos e (ii) sempre analisar os pontos de código que triggam a alteração de estado de um objeto (`view.addSubview(collectionView)` por exemplo) e se questionar se esta linha está em um ponto do código em que esta alteração de estado seja razoável ou faça sentido (em outras palavras, onde seja obvio pra outro programador ou você do futuro. Exemplo: se o caso é a necessidade de adicionar uma collection view ao WorkspaceViewController, então o ponto onde a linha `view.addSubview(collectionView)` é permitida é em WorkspaceViewController.swift. Contrariar essa premissa pode te levar a dificuldades de rastrear o estado da aplicação como um todo como citado antes, e levar a mais bugs).